crates/client/src/storage/fs.rs

```diff
@@ -42,10 +47,13 @@ impl FileSystemRegistryStorage {
     /// If the lock cannot be acquired, `Ok(None)` is returned.
     pub fn try_lock(base_dir: impl Into<PathBuf>) -> Result<Option<Self>> {
         let base_dir = base_dir.into();
+        let mut registries_dir = base_dir.clone();
+        registries_dir.pop();
```

Comment:
This can be silently incorrect if `base_dir` is empty.

```suggestion
        let registries_dir = base_dir.parent()
            .context("base_dir cannot be empty")?
            .to_path_buf();
        registries_dir.pop();
```